llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp

```diff
@@ -2229,6 +2230,112 @@ void DwarfDebug::beginFunctionImpl(const MachineFunction *MF) {
   // Record beginning of function.
   PrologEndLoc = emitInitialLocDirective(
       *MF, Asm->OutStreamer->getContext().getDwarfCompileUnitID());
+
+  // For each MBB we try to determine whether and what source line is *always*
+  // the last stepped-on line before entering MBB. Such a line exists if every
```

Comment:
This is nitpicking, but I feel the order of this sentence should be flipped to make it clear at the start what's going on -- i.e. "Find blocks where the line stepped on before entering it isn't always the same". Otherwise it's quite oblique to work out what the subject is, until the end.